pynemo/tide/nemo_bdy_tide3.py

```diff
@@ -1,272 +1,291 @@
+                    logger.error(' Error: Land or Mask contamination')
+
+                cosuX[h, :] = ampuX[c, :] * np.cos(np.deg2rad(phauX[c, :])) / depu
+                sinuX[h, :] = ampuX[c, :] * np.sin(np.deg2rad(phauX[c, :])) / depu
+                cosvX[h, :] = ampvX[c, :] * np.cos(np.deg2rad(phavX[c, :])) / depu
+                sinvX[h, :] = ampvX[c, :] * np.sin(np.deg2rad(phavX[c, :])) / depu
+                cosuY[h, :] = ampuY[c, :] * np.cos(np.deg2rad(phauY[c, :])) / depv
+                sinuY[h, :] = ampuY[c, :] * np.sin(np.deg2rad(phauY[c, :])) / depv
+                cosvY[h, :] = ampvY[c, :] * np.cos(np.deg2rad(phavY[c, :])) / depv
+                sinvY[h, :] = ampvY[c, :] * np.sin(np.deg2rad(phavY[c, :])) / depv
+            else:
+                cosuX[h, :] = 0.01 * ampuX[c, :] * np.cos(np.deg2rad(phauX[c, :]))
+                sinuX[h, :] = 0.01 * ampuX[c, :] * np.sin(np.deg2rad(phauX[c, :]))
+                cosvX[h, :] = 0.01 * ampvX[c, :] * np.cos(np.deg2rad(phavX[c, :]))
+                sinvX[h, :] = 0.01 * ampvX[c, :] * np.sin(np.deg2rad(phavX[c, :]))
+                cosuY[h, :] = 0.01 * ampuY[c, :] * np.cos(np.deg2rad(phauY[c, :]))
+                sinuY[h, :] = 0.01 * ampuY[c, :] * np.sin(np.deg2rad(phauY[c, :]))
+                cosvY[h, :] = 0.01 * ampvY[c, :] * np.cos(np.deg2rad(phavY[c, :]))
+                sinvY[h, :] = 0.01 * ampvY[c, :] * np.sin(np.deg2rad(phavY[c, :]))
+
+# TOD:: Do we need to rotate ??? And is this method  correct ????
+    maxJ = DC.lonlat['t']['lon'].shape[0]
+    maxI = DC.lonlat['t']['lon'].shape[1]
+    dst_gcos = np.ones([maxJ, maxI])
+    dst_gsin = np.zeros([maxJ, maxI])
+    #lets start with the u-points
+    grid_angles = nemo_bdy_grid_angle.GridAngle(setup.settings['dst_hgr'], 0, maxI, 0, maxJ, 'u')
+    dst_gcos = grid_angles.cosval
+    dst_gsin = grid_angles.sinval
+
+    #retain only boundary points rotation information
+    tmp_gcos = np.zeros((Grid_U.bdy_r==0).sum())
+    tmp_gsin = np.zeros((Grid_U.bdy_r==0).sum())
+    for index in range(len(tmp_gcos)):
+        tmp_gcos[index] = dst_gcos[Grid_U.bdy_i[index, 1], Grid_U.bdy_i[index, 0]] # Iterate over bdy_i (not bdy_r )because the
+        tmp_gsin[index] = dst_gsin[Grid_U.bdy_i[index, 1], Grid_U.bdy_i[index, 0]] #  first portion are the edge values
+    dst_gcos = tmp_gcos
+    dst_gsin = tmp_gsin
+
+    cosu = rot_rep(cosuX, cosvX, 'u', 'en to i', dst_gcos, dst_gsin)
+    sinu = rot_rep(sinuX, sinvX, 'u', 'en to i', dst_gcos, dst_gsin)
+
+    #let do the v points
+    dst_gcos = np.ones([maxJ, maxI])
+    dst_gsin = np.zeros([maxJ, maxI])
+    grid_angles = nemo_bdy_grid_angle.GridAngle(setup.settings['dst_hgr'], 0, maxI, 0, maxJ, 'v')
+    dst_gcos = grid_angles.cosval
+    dst_gsin = grid_angles.sinval
+
+    #retain only boundary points rotation information
+    tmp_gcos = np.zeros((Grid_V.bdy_r==0).sum())
+    tmp_gsin = np.zeros((Grid_V.bdy_r==0).sum())
```

Comment:
As before, force the size of the `tmp_`trig variables to be of size equivalent to when `rim_width=1`